stdlib/internal/promql/join.md

```diff
@@ -0,0 +1,91 @@
+### join
+
+`join` merges two input streams into a single output stream.
+Records that have the same group key and `_time` values will be joined in the output.
+The output groups will be the same as the input groups.
+
+`join` has the following properties:
+
+| Name    | Type     | Description |
+| ------- | -------- | ----------- |
+| tables  | object   | Specifies the two table streams to join. |
+| fn      | function | Defines the function that joins the rows of each table. The return value is an object which defines the output record structure. |
+
```

Comment:
You could also add a `predFn` parameter that would define the join predicate, and make it default to `(left, right) => left._time == right._time`.

Reply:
Yes I agree, that's ultimately how I would like the Flux join function defined. However that will also require us to provide some internal join implementations that are optimized for specific scenarios for which the planner will need to be made aware. I didn't want this implementation to have the overhead of calling an anonymous function for each row combination so I left that out.